src/Extension/Mention/MentionParser.php

```diff
@@ -70,7 +70,7 @@ public function parse(string $match, InlineParserContext $inlineContext): bool
         $previousState = $cursor->saveState();
 
         // Advance past the symbol to keep parsing simpler
-        $cursor->advance();
+        $cursor->advanceBy(\mb_strlen($this->symbol));
 
         // Parse the identifier
         $identifier = $cursor->match($this->mentionRegex);
```

Comment:
I feel like both the symbol and mention regex should be combined into a single regex that uses parenthesis to group the individual results. Maybe something like:

```php
        return InlineParserMatch::join(
          InlineParserMatch::string($this->symbol),
          InlineParserMatch::regex($this->mentionRegex),
        );
```

Unfortunately, this would also mean that `string $match` would have to really be `array $matches`.

Also, is there a reason we can't add both of these to `InlineParserContext`, i.e. `$inlineContext->getMatch(): string` and `$inlineContext->getMatches(): array`; the former just returnining the first item of the latter?

Then we could something like the following here:

```php
        // Extract the symbol and identifier from the matches.
        [$symbol, $identifier] = $inlineContext->getMatches();

        // Advance past the symbol to keep parsing simpler
        $cursor->advanceBy(\mb_strlen($symbol));
```

Reply:
Great idea!  I've incorporated this into #560.